### PR TITLE
Show pins as list and display on map

### PR DESCRIPTION
**Pull request by naglepuff**
This PR adds some basic functionality to pins. To demonstrate the UI changes, three pin objects are created with the `populate` command. These pins just contain notes and do not point to child dataset objects.

To test these changes, run the populate script to ensure you have these pins. In front-end application, navigate to the `InvestigationDetail` view for the "Whole Team Investigation." In the sidebar you should see a `Pins` tab, which lists all pins for the investigation, including the three pins added to the populate script as part of these changes. Click the checkbox next to any of these pins to toggle a geojs feature for that pin. It should display with the pin's color. This can be checked from the admin console.

Hover over the pin on the geojs map. The note should appear at your mouse. Moving the mouse outside of the pin will cause the note to disappear again.

Test selecting/deselecting pins in the sidebar. Test collapsing/un-collapsing the sidebar. For all of these tasks, verify that the selected pins in the sidebar reflect the pins shown on the map.

These changes relate to #71 and #92, but does not fully address those issues.

### CODE REVIEW

File: web/src/views/InvestigationDetail.vue [hunk 1]
```diff
@@ -151,17 +172,82 @@ export default defineComponent({
       geojsParams.layer.crossDomain = 'use-credentials';
       createMap(geojsParams.map);
       createLayer('osm', geojsParams.layer);
+      clampBoundsX(false);
+    });
+    // #endregion datasets
+
+    // #region pin
```
Comment (waxlamp) [inline web/src/views/InvestigationDetail.vue]: Is this like a folding support for an IDE or something?

File: web/src/views/InvestigationDetail.vue [hunk 2]
```diff
@@ -151,17 +172,82 @@ export default defineComponent({
...
+    /* eslint-enable */
+    watch(selectedPins, (pinList) => {
+      if (!featureLayer) {
+        featureLayer = createLayer('feature', { features: ['point', 'line', 'polygon'] });
+      }
+      const pinFeatureData = pinList.map((pin) => {
+        let pinLocation: Point | undefined = postGisToPoint(pin.child_location);
+        if (!pinLocation) {
+          pinLocation = { x: 0, y: 0 };
+        }
```
Comment (waxlamp) [inline web/src/views/InvestigationDetail.vue]: I think the following should work (but I haven't tested it):

```suggestion
        const pinLocation: Point = postGisToPoint(pin.child_location) || { x: 0, y: 0 };
```

File: web/src/views/InvestigationDetail.vue [hunk 3]
```diff
@@ -151,17 +172,82 @@ export default defineComponent({
...
+          note: pin.note,
+        };
+      });
+      if (!pinFeature) {
+        /* eslint-disable */
+        pinFeature = featureLayer.createFeature('point')
+          .data(pinFeatureData)
+          .position((pin: any) => ({ x: pin.x, y: pin.y }))
+          .style({
+            // reasonable default?
```
Comment (waxlamp) [inline web/src/views/InvestigationDetail.vue]: Yes, this is fine.

```suggestion
```

**Review by waxlamp — APPROVED** [review]
Good stuff. Lots here to build off of!

File: web/src/views/InvestigationDetail.vue (hunk 1 above)
Comment (naglepuff) [reply web/src/views/InvestigationDetail.vue]: Yeah, it is. I was not thinking of the fact that there's no standard tool people use. I don't mind removing the region markers if that's best practice.

Comment (waxlamp) [reply web/src/views/InvestigationDetail.vue]: It's probably best to remove them.

What IDE are you using? I know vim does this sort of thing in-band but in modern editors you ought to be able to do it out-of-band (so to speak).

Comment (naglepuff) [reply web/src/views/InvestigationDetail.vue]: vscode

Comment (naglepuff) [reply web/src/views/InvestigationDetail.vue]: Removed in e8f054f

**Review by mcovalt — APPROVED** [review]

**Pull request merged**